storm-core/src/jvm/backtype/storm/topology/WindowedBoltExecutor.java

```diff
@@ -0,0 +1,224 @@
+        } else if (windowLengthCount != null) {
+            ensureCountLessThanMaxPending(windowLengthCount.value, maxSpoutPending);
+        } else if (slidingIntervalCount != null) {
+            ensureCountLessThanMaxPending(slidingIntervalCount.value, maxSpoutPending);
+        }
+    }
+
+    private WindowManager<Tuple> initWindowManager(WindowLifecycleListener<Tuple> lifecycleListener, Map stormConf) {
+        WindowManager<Tuple> manager = new WindowManager<>(lifecycleListener);
+        Duration windowLengthDuration = null;
+        Count windowLengthCount = null;
+        Duration slidingIntervalDuration = null;
+        Count slidingIntervalCount = null;
+        if (stormConf.containsKey(Config.TOPOLOGY_BOLTS_WINDOW_LENGTH_COUNT)) {
+            windowLengthCount = new Count(((Number) stormConf.get(Config.TOPOLOGY_BOLTS_WINDOW_LENGTH_COUNT)).intValue());
+        } else if (stormConf.containsKey(Config.TOPOLOGY_BOLTS_WINDOW_LENGTH_DURATION_MS)) {
+            windowLengthDuration = new Duration(
+                    ((Number) stormConf.get(Config.TOPOLOGY_BOLTS_WINDOW_LENGTH_DURATION_MS)).intValue(),
+                    TimeUnit.MILLISECONDS);
+        }
+
+        if (stormConf.containsKey(Config.TOPOLOGY_BOLTS_SLIDING_INTERVAL_COUNT)) {
+            slidingIntervalCount = new Count(((Number) stormConf.get(Config.TOPOLOGY_BOLTS_SLIDING_INTERVAL_COUNT)).intValue());
+        } else if (stormConf.containsKey(Config.TOPOLOGY_BOLTS_SLIDING_INTERVAL_DURATION_MS)) {
+            slidingIntervalDuration = new Duration(((Number) stormConf.get(Config.TOPOLOGY_BOLTS_SLIDING_INTERVAL_DURATION_MS)).intValue(), TimeUnit.MILLISECONDS);
+        } else {
+            // default is a sliding window of count 1
+            slidingIntervalCount = new Count(1);
+        }
+        // validate
+        validate(stormConf, windowLengthCount, windowLengthDuration,
+                 slidingIntervalCount, slidingIntervalDuration);
+        if (windowLengthCount != null) {
+            manager.setWindowLength(windowLengthCount);
+        } else {
+            manager.setWindowLength(windowLengthDuration);
+        }
+        if (slidingIntervalCount != null) {
+            manager.setSlidingInterval(slidingIntervalCount);
+        } else {
+            manager.setSlidingInterval(slidingIntervalDuration);
+        }
+        return manager;
+    }
+
+    @Override
+    public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
+        this.windowedOutputCollector = new WindowedOutputCollector(collector);
+        bolt.prepare(stormConf, context, windowedOutputCollector);
+        this.listener = newWindowLifecycleListener();
+        this.windowManager = initWindowManager(listener, stormConf);
+        LOG.debug("Initialized window manager {} ", this.windowManager);
+    }
+
+    @Override
+    public void execute(Tuple input) {
+        windowManager.add(input);
```

Comment:
I added it with the idea of supporting late events, but realized that it will need more work and thought it can be added once the basic windowing is in place. I will file a follow on jira and work on it.